perl/Makefile

```diff
@@ -0,0 +1,83 @@
+GOOD_FEATURE_FILES = $(shell find ../testdata/good -name "*.feature")
+BAD_FEATURE_FILES  = $(shell find ../testdata/bad -name "*.feature")
+
+TOKENS   = $(patsubst ../testdata/%.feature,acceptance/testdata/%.feature.tokens,$(GOOD_FEATURE_FILES))
+ASTS     = $(patsubst ../testdata/%.feature,acceptance/testdata/%.feature.ast.json,$(GOOD_FEATURE_FILES))
+PICKLES  = $(patsubst ../testdata/%.feature,acceptance/testdata/%.feature.pickles.json,$(GOOD_FEATURE_FILES))
+ERRORS   = $(patsubst ../testdata/%.feature,acceptance/testdata/%.feature.errors,$(BAD_FEATURE_FILES))
+
+PERL_FILES = $(shell find . -name "*.pm")
+
+all: .compared
+.PHONY: all
+
+.compared: .built $(TOKENS) $(ASTS) $(PICKLES) $(ERRORS)
+	touch $@
+
+.cpanfile_dependencies:
+	cpanm --installdeps .
+	touch $@
+
+.built: lib/Gherkin/Generated/Parser.pm lib/Gherkin/Generated/Languages.pm $(PERL_FILES) bin/gherkin-generate-tokens bin/gherkin-generate-ast LICENSE.txt .cpanfile_dependencies
+	@$(MAKE) --no-print-directory show-version-info
+	# add Perl-level unit tests
+	touch $@
+
+show-version-info:
+	perl --version
+.PHONY: show-version-info
+
+acceptance/testdata/%.feature.tokens: ../testdata/%.feature ../testdata/%.feature.tokens .built
+	mkdir -p `dirname $@`
+	bin/gherkin-generate-tokens $< > $@
+	diff --unified $<.tokens $@
+.DELETE_ON_ERROR: acceptance/testdata/%.feature.tokens
+
+acceptance/testdata/%.feature.ast.json: ../testdata/%.feature ../testdata/%.feature.ast.json .built
+	mkdir -p `dirname $@`
+	bin/gherkin-generate-ast $< > $@
+	diff --unified $<.ast.json $@
+.DELETE_ON_ERROR: acceptance/testdata/%.feature.ast.json
+
+acceptance/testdata/%.feature.pickles.json: ../testdata/%.feature ../testdata/%.feature.pickles.json .built
+	mkdir -p `dirname $@`
+	bin/gherkin-generate-pickles $< > $@
+	diff --unified $<.pickles.json $@
+.DELETE_ON_ERROR: acceptance/testdata/%.feature.pickles.json
+
+acceptance/testdata/%.feature.errors: ../testdata/%.feature ../testdata/%.feature.errors .built
+	mkdir -p `dirname $@`
+	! bin/gherkin-generate-ast $< 2> $@
+	diff --unified $<.errors $@
+.DELETE_ON_ERROR: acceptance/testdata/%.feature.errors
+
+gherkin/gherkin-languages.json: ../gherkin-languages.json
```

Comment:
This target fails with:

```
cp ../gherkin-languages.json gherkin/gherkin-languages.json
cp: gherkin/gherkin-languages.json: No such file or directory
make: *** [gherkin/gherkin-languages.json] Error 1
```

I think it would be better if `helper-scripts/build_languages.pl` read `../gherkin-languages.json` directly (via STDIN). No need to copy the file if it's just a temporary copy.